src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -323,11 +323,20 @@ internal void Update(double number)
                             }
                         }
 
-                        lock (this.histogramBuckets.LockObject)
+                        var sw = default(SpinWait);
+                        while (true)
                         {
-                            this.runningValue.AsLong++;
-                            this.histogramBuckets.RunningSum += number;
-                            this.histogramBuckets.RunningBucketCounts[i]++;
+                            if (Interlocked.Exchange(ref this.histogramBuckets.UsingHistogram, 1) == 0)
+                            {
+                                this.runningValue.AsLong++;
```

Comment:
I haven't look into the underlying code, wonder if this `++` might throw (e.g. integer overflow case). If that's the case, we might need to make sure we release the spinlock.

Reply:
We don't acquire any lock in the first place to be released. `SpinWait` is used to just smartly apply context switch for the thread.

Reply:
one Ln335 we set `this.histogramBuckets.UsingHistogram = 0`, if we throw before this, all other threads would spin I guess?

Reply:
I think `unsafe` would solve the problem here.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/CircularBuffer.cs#L139-L143 We could explore if we need to have an exit plan like done for CircularBuffer.